File: web/src/types/kit.ts
export interface KitSummary {
  id: string
  name: string
  version: string
  description: string
  tags: string[]
  author: string
  lastUpdated?: string
}

export interface KitFile {
  path: string
  content: string
  lastUpdated?: string
}

export interface Variable {
  name: string
  type: string
  default: string
  description: string
  required: boolean
}

export interface KitDetail extends KitSummary {
  files: KitFile[]
  readme: string
  variables: Variable[]
}

export interface KitsResponse {
  kits: KitSummary[]
  lastSync?: string
}

export interface SyncResponse {
  status: string
  lastSync?: string
}

File: internal/domain/models/kit.go
package models

type Kit struct {
	ID          string     `json:"id"`
	Name        string     `json:"name"`
	Version     string     `json:"version"`
	Author      string     `json:"author"`
	Description string     `json:"description"`
	Files       []File     `json:"files"`
	Variables   []Variable `json:"variables"`
	Readme      string     `json:"readme"`
	Tags        []string   `json:"tags"`
}
